src/axom/inlet/tests/inlet_object.cpp

```diff
@@ -990,6 +993,7 @@ TYPED_TEST(inlet_object, default_struct_field_marked_true)
   // The container itself exists and was provided by the user
   EXPECT_TRUE(foo_container.exists());
   EXPECT_TRUE(foo_container.isUserProvided());
+  EXPECT_TRUE(inlet.isUserProvided("foo"));
```

Comment:
Consider also adding a check for the following with comments?
* `isUserProvided("bar")` 
* `isUserProvided("baz")` 
* `isUserProvided("foo/bar")` 
* `isUserProvided("foo/baz")` 
* `isUserProvided("qux")` 

Based on the examples and comments, I'm not entirely clear on which entities will return `true` for `isUserProvided`.  
I think `foo/bar` would be `true`, and the rest `false`, but I'm not entirely sure about the former.

Reply:
Thanks for updating the tests @joshessman-llnl 

My only remaining question is about how this would work if the test string has a field that is not in the schema.  E.g.  the input is ` foo = {quux = true }`, but the schema is `foo.bar`.
Would  `foo.isUserContained("quux")` be true ?

Reply:
It would not - Inlet's logic only calls out to the reader when something is added to the schema, so neither `contains` nor `isUserProvided` would return true for "quux".  However, the `unexpectedNames` logic added a little while back would include "quux".